metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -1781,6 +1782,7 @@ public CompletableFuture<Void> unregisterBroker(
         ControllerRequestContext context,
         int brokerId
     ) {
+        controllerMetrics.removeTimeSinceLastHeartbeatMetric(brokerId);
```

Comment:
In order to prevent the wrong thing getting unregistered, it seems like we should check the broker epoch here.